src/framework/uicomponents/qml/Muse/UiComponents/StyledMenuLoader.qml

```diff
@@ -74,7 +74,9 @@ Loader {
         }
 
         onClosed: function(force) {
-            Qt.callLater(prv.unloadMenu, force)
+            // Not using Qt.callLater here because the delayed call causes
+            // a crash if objects are destroyed in the meantime.
+            prv.unloadMenu(force)
```

Comment:
can we reproduce this crash on latest master or did this start happening after your changes?

Reply:
I am easily able to reproduce it in 4.4.1 (without my changes). Give it a try yourself. :)

Reply:
I am wondering what if we listen for the onDestroy event of the menu and handle it properly?

Reply:
I think I was able to solve this more elegantly. In PaletteGridView.qml the parent of the menu loader was getting set to the target cell otherwise the context menu would not be positioned correctly. I started wondering why this was the case since the right-click area is now attached not to the cells but to the paletteView so the parent had to be paletteView in order for the mouse coordinates provided in the click event to position the menu correctly. I discovered that StyledMenuLoader.update(model, x = -1, y = -1) wasn't working properly: it first set the x and y properties of the menu and at the end called **_menu.calculateSize()_** which at the end was setting x to 0 and y to the height of the parent. This effectively was ignoring the (x,y) coordinates passed to StyledMenuLoader.update. I moved the call to **_menu.calculateSize()_** earlier and everything started working properly. Then I was able to set the parent of the context menu loader not to the target cell but to the paletteView. I am not sure why this caused the crash to go away even with Qt.callLater so I brought it back. I am assuming that paletteView does not get destroyed when activating search mode unlike the cells.
Is there any reason why **_menu.calculateSize()_** sets x = 0 and y = parent.height? Or why MuseScore Studio shows context menus not under the mouse (if possible) but at (0, parent.height) so below the cell and left-aligned with it? What do you guys think? @Eism @RomanPudashkin @cbjeukendrup

Reply:
Gah. If the context menu cannot appear at the mouse coordinates due to lack of space, it will be automatically repositioned above the parent. So the parent must be the cell clicked and not the entire palette.

Reply:
I've looked at how the context menu on the tabs is implemented and saw some interesting things there... 😃

Reply:
I've changed the code to use ContextMenuLoader like the context menu of tabs does. I think it looks good and works well. Guys, please take a look. Thanks!